### PR TITLE
Multiple layers

### LINKED ISSUE #1036: if you have multiple layers, and then query an edge without specifying the layer, it returns none
g.edge should search all layers by default 

g = Graph ()
g.add_edge(1,1,2,{"tx_sent":10},layer="btc")
g.add_edge(1,1,2,{"tx_sent":20},layer="eth")
g.add_edge(1,1,2,{"tx_sent":70},layer="tether")

e = g.layer(["eth","btc"]).edge(1,2) # this returns 1 edge view

e.property("tx_sent").temporal().sum() == 30

e_list = g.layer("btc,eth").edges() 
len(e_list) == 1

e = g.edge(1,2)
e_btc = e.layer("btc")
e_eth = e.layer("eth")
e_btc.property("tx_sent").temporal().sum() < e_eth.property("tx_sent").temporal().sum()

e_eth = e_eth.layer(["eth","btc"])
e_eth.property("tx_sent").temporal().sum() == 30

e = g.edge(1,2)
e.property("tx_sent").temporal().sum() == 100

### PR DESCRIPTION
**Pull request by fabianmurariu**
### Issues

fixes #1036

### CODE REVIEW

File: raphtory/src/db/internal/core_ops.rs [hunk 1]
```diff
@@ -99,19 +102,16 @@ impl<const N: usize> CoreGraphOps for InnerTemporalGraph<N> {
 
     fn static_edge_prop(&self, e: EdgeRef, name: &str) -> Option<Prop> {
```
Comment (ljeub-pometry) [inline raphtory/src/db/internal/core_ops.rs:103]: This is indeed very broken, we need to figure out merging properties across layers!

File: raphtory/src/python/graph/edge.rs [hunk 2]
```diff
@@ -312,6 +315,28 @@ impl PyEdge {
...
     }
 
+    /// Get a new Edge with the properties of this Edge within the specified layers.
+    ///
+    /// Arguments:
+    ///   layer_names ([str]): Layers to be included in the new edge.
+    ///
+    /// Returns:
+    ///   A new Edge with the properties of this Edge within the specified time window.
+    #[pyo3(signature = (layer_names))]
```
Comment (ljeub-pometry) [inline raphtory/src/python/graph/edge.rs:325]: check this signature makes sense

File: raphtory/src/python/graph/edge.rs [hunk 3]
```diff
@@ -312,6 +315,28 @@ impl PyEdge {
...
+    pub fn layers(
+        &self,
+        layer_names: Vec<String>,
+    ) -> PyResult<EdgeView<LayeredGraph<DynamicGraph>>> {
+        if let Some(edge) = self.edge.layer(layer_names.clone().into()) {
```
Comment (ljeub-pometry) [inline raphtory/src/python/graph/edge.rs:330]: Should this return error or option?

**Review by ljeub-pometry — CHANGES_REQUESTED** [review]
We need to fix handling of static properties and check/fix the materialize functionality before we can merge this

File: raphtory/src/db/api/view/graph.rs [hunk 4]
```diff
@@ -188,7 +185,6 @@ impl<G: BoxableGraphView + Sized + Clone> GraphViewOps for G {
     fn get_unique_layers(&self) -> Vec<String> {
         self.get_unique_layers_internal()
             .into_iter()
-            .filter(|id| *id != 0) // the default layer has no name
             .map(|id| self.get_layer_name_by_id(id))
```
[the remark this answers is not in the document]
Comment (fabianmurariu) [reply raphtory/src/db/api/view/graph.rs:193]: why not?

File: examples/rust/src/bin/crypto/main.rs [hunk 5]
```diff
@@ -38,13 +38,13 @@ fn main() {
     println!("Time taken: {} secs", now.elapsed().as_secs());
 
     let now = Instant::now();
-    let _ = unweighted_page_rank(&g.layer("USDT").unwrap(), 20, None, None, true);
+    let _ = unweighted_page_rank(&g.layer("USDT".into()).unwrap(), 20, None, None, true);
```
[the remark this answers is not in the document]
Comment (fabianmurariu) [reply examples/rust/src/bin/crypto/main.rs]: I suppose I could make it take `AsRef<str>`

[the remark this answers is not in the document]
Comment (ljeub-pometry) [reply examples/rust/src/bin/crypto/main.rs]: Made this generic now

File: raphtory/src/db/api/view/graph.rs [hunk 6]
```diff
@@ -323,27 +306,34 @@ impl<G: BoxableGraphView + Sized + Clone> GraphViewOps for G {
...
+                    )?;
+                }
             }
             if self.include_deletions() {
-                for t in self.edge_deletion_history(e.edge) {
-                    g.delete_edge(t, e.src().id(), e.dst().id(), layer)?;
+                for t in self.edge_deletion_history(e.edge, LayerIds::All) {
+                    for layer in layer_names.iter() {
+                        g.delete_edge(t, e.src().id(), e.dst().id(), Some(layer))?;
+                    }
```
[the remark this answers is not in the document]
Comment (ljeub-pometry) [reply raphtory/src/db/api/view/graph.rs]: fixed

File: raphtory/src/db/mod.rs.orig [hunk 7]
```diff
@@ -0,0 +1,12 @@
+pub mod api;
+pub mod graph;
+pub(crate) mod internal;
+pub mod task;
+<<<<<<< HEAD
```
Comment (Haaroon) [inline raphtory/src/db/mod.rs.orig:5]: Issue in merge

**Review by Haaroon — APPROVED** [review]

**Pull request merged**